### DISCUSSION TITLE
Is there a way to use a 16-color theme for xonsh, or something close to it?

### DISCUSSION THREAD

**Discussion by ilyagr** [xonsh/xonsh#5703] [Q&A]
I'd like xonsh to match my terminal palette. I couldn't find anything like `ansi` color theme, and the `native` color scheme looks like it's using different colors. I also tried the (still documented) `PROMPT_TOOLKIT_COLOR_DEPTH` environment variable, but it seemed to do nothing, and https://github.com/xonsh/xonsh/issues/854 says it's obsolete.

**Comment by anki-code** [comment]
Maybe this can help - https://github.com/ericktucto/xontrib-base16-shell from [xontribs](https://github.com/topics/xontrib).

**Reply by ilyagr** [reply]
I am pretty sure base16 shells is something slightly different, despite the name it seems to be about colors 17-21: https://github.com/chriskempson/base16-shell.

It seems like something I might like, but at the moment I don't use it and don't understand how to use it. How should I add colors 17-21 to [my existing 16 shell colors](https://github.com/ilyagr/dotfiles/blob/8d7bac0bc4d3dd2f8098926c6a682187616870b4/.config/kitty/gruvbox-material-dark.conf#L63-L93)?

**Reply by anki-code** [reply]
Maybe @dyuri can help with this.

**Reply by dyuri** [reply]
We are using PTK/pygments styles, and the predefined ones are quite a mess imho.
You can define your own though, redefining the tokens (see `xonfig colors`): https://xon.sh/customization.html#registering-custom-styles
Related PTK documentation: https://python-prompt-toolkit.readthedocs.io/en/stable/pages/advanced_topics/styling.html

**Comment by marchank0** [comment]
@ilyagr did you make any progress?
